src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerCleanerTest.java

```diff
@@ -73,6 +73,18 @@ void should_not_find_any_blobs() {
         verifyNoInteractions(containerClient);
     }
 
+    @Test
+    void should_handle_repository_exception() {
+        // given
+        given(envelopeRepository.find(DISPATCHED, false)).willThrow(new RuntimeException("msg"));
+
+        // when
+        containerCleaner.process(CONTAINER_NAME);
+
+        // then
+        verifyNoInteractions(containerClient);
+    }
```

Comment:
perhaps `ContainerCleaner` could remain as it is in this PR?

Reply:
This change is related - we want to try to process all containers, regardless of errors.

Reply:
maybe do try-catch in the job, when it's calling this service

Reply:
following the 'throw early catch late' principle